src/main/java/org/plumelib/util/StringsPlume.java

```diff
@@ -1049,24 +1049,44 @@ public static String mapToStringAndClass(Map<?, ?> m) {
 
   /**
    * Return either "n <em>noun</em>" or "n <em>noun</em>s" depending on n. Adds "es" to words ending
-   * with "ch", "s", "sh", or "x".
+   * with "ch", "s", "sh", or "x", adds "ies" to words ending with "y" when the previous letter is
+   * consonant, adds "ves" to words ending in "f" or "fe".
    *
    * @param n count of nouns
    * @param noun word being counted
    * @return noun, if n==1; otherwise, pluralization of noun
    */
   @SideEffectFree
   public static String nplural(int n, String noun) {
+    if (noun.length() < 2) {
+      return "null";
+    }
     if (n == 1) {
       return n + " " + noun;
-    } else if (noun.endsWith("ch")
-        || noun.endsWith("s")
-        || noun.endsWith("sh")
-        || noun.endsWith("x")) {
+    }
+    char lastLetter = noun.charAt(noun.length() - 1);
+    char penultimateLetter = noun.charAt(noun.length() - 2);
+    if ((penultimateLetter == 'c' && lastLetter == 'h')
+        || lastLetter == 's'
+        || (penultimateLetter == 's' && lastLetter == 'h')
+        || lastLetter == 'x') {
       return n + " " + noun + "es";
-    } else {
-      return n + " " + noun + "s";
     }
+    if (lastLetter == 'y'
+        && (penultimateLetter != 'a'
+            && penultimateLetter != 'e'
+            && penultimateLetter != 'i'
+            && penultimateLetter != 'o'
+            && penultimateLetter != 'u')) {
+      return n + " " + noun.substring(0, noun.length() - 1) + "ies";
+    }
+    if (lastLetter == 'f') {
```

Comment:
The rules about "f" and "fe" seem like exceptions rather than the standard case.  I examined words starting with a-e in a dictionary.

Of those, 4 change "f" to "v" in the plural:  afterlife bookshelf calf elf

By contrast, 19 plurals retain the "f":  aperitif bailiff belief blastoff bluff  brief carafe castoff chef chief clef cliff cloverleaf coif cuff cutoff disproof distaff earmuff

One plural can be written either way:  dwarf

Therefore, I think the rules about "f" should be removed.  Clients have to deal with irregular words on their own, but the method should handle the common case.

Reply:
To be honest, i did not make a big research although i knew that there are exceptions, because for some reason i thought that there are more nouns that follow the rule than the other way around. There are of course some other nouns that follow the rule like wolf and leaf but if there are 19 that they don't, then it makes sense to remove this rule.

Reply:
The point is not that there are 19 words that use the standard rule rather than the special case.  The point is that 82% of the words starting with a-e use the standard rule rather than the special case, and so probably most English words do.  I think the code should work for the majority of English words, even if it is wrong for some words.

Reply:
Yes i agree. I just didn't realise that most nouns do not follow this rule.